src/pages/movie-page/movie-page.tsx

```diff
@@ -0,0 +1,15 @@
+import { useParams } from 'react-router-dom';
+import { useState } from 'react';
+import { Movie } from '../../types/movie';
+import MovieInfo from '../../components/movie-info/movie-info';
+
+export default function MoviePage() {
+  const { movieId } = useParams();
+  const [movie, setMovie] = useState<Movie | null>(null);
+
+  fetch(`http://localhost:4000/movies/${movieId}`)
```

Comment:
It's bad practice to put loading into the render because if a component rerenders, it will load another film.

1. Let's create a service (move this request to movie service).
2. Let's create a custom hook (useMovie).
3. If the film loads 30 sec it's better to show some loading to the user.
4. Server errors should be handled somehow.

Reply:
Added react router loader and errorElement to handle loading and errors.